root/io/autoptr/Event_v2.hxx

```diff
@@ -0,0 +1,36 @@
+#ifndef EVENT_V2_H
+#define EVENT_V2_H
+
+#include <RtypesCore.h>
+
+#include <memory>
+
+// Since std::auto_ptr is deprecated and may not exist anymore, we emulate it.
+// This class is not actually emulating the auto_ptr behavior. It is a placeholder for this test.
+template <typename T>
+struct EmulatedAutoPtr {
+   T *fRawPtr = nullptr;
+
+   EmulatedAutoPtr() = default;
+   explicit EmulatedAutoPtr(T *rawPtr) : fRawPtr(rawPtr) {}
+   EmulatedAutoPtr &operator=(const EmulatedAutoPtr &other)
+   {
+      fRawPtr = other.fRawPtr;
+      return *this;
+   }
```

Comment:
```suggestion
   EmulatedAutoPtr &operator=(EmulatedAutoPtr &other)
   {
      fRawPtr = other.fRawPtr;
      other.fRawPtr = nullptr;
      return *this;
   }
```
is a simple change that remove the "not actually emulating" comment ;)

Reply:
(requires adjustment in the writing macro)

Reply:
I am not sure I understand: what is missing to merge the PR?

Reply:
I was just pointing out that I had to change more code than Philippe's suggestion to implement the change.